source/read/count.txt

```diff
@@ -0,0 +1,222 @@
+.. _cpp-count:
+
+===============
+Count Documents
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: number, amount, estimation, code example
+
+Overview
+---------
+
+In this guide, you can learn how to use the {+driver-short+} to retrieve an accurate
+and estimated count of the number of documents in a collection. The ``count_documents()``
+method returns the exact number of documents that match a query filter or that exist in
+a collection, and the ``estimated_document_count()`` method returns the estimated number
+of documents in a collection.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``companies`` collection in the ``sample_training``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your C++ application, instantiate a ``mongocxx::client`` that connects to an Atlas
+cluster and assign the following values to your ``db`` and ``collection`` variables:
+
+.. literalinclude:: /includes/read/count.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<cpp-get-started>` tutorial.
+
+.. _cpp-accurate-count:
+
+Retrieve an Accurate Count
+--------------------------
+
+Use the ``count_documents()`` method to count the number of documents that are in a
+collection. To count the number of documents that match a specific search
+critera, pass a query filter document to the ``count_documents()`` method.
+
+To learn more about specifying a query, see the :ref:`cpp-specify-query` guide.
+
+Count All Documents
+~~~~~~~~~~~~~~~~~~~
+
+To return a count of all documents in the collection, pass an empty filter document to
+the ``count_documents()`` method, as shown in the following example:
+
+.. code-block:: cpp
+
+   auto count = collection.count_documents({})
```

Comment:
Q: why not include this in the code file as well? Also why not include a print statement?

Reply:
moved to the code file & added output - it seemed like a basic syntax example, but I think it still benefits from those changes